mam/v2/defs.h

```diff
@@ -0,0 +1,206 @@
+
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * MAM is based on an original implementation & specification by apmi.bsu.by
+ [ITSec Lab]
+
+ *
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+/*!
+\file defs.h
+\brief MAM2 common definitions.
+*/
+#ifndef __MAM_V2_DEFS_H__
+#define __MAM_V2_DEFS_H__
+
+/*! \brief Enable debug output macros. */
+//#define MAM2_DEBUG
+
+#include <stddef.h>
+#include <stdint.h>
+
+#if defined(NDEBUG)  //&& defined(MAM2_DEBUG)
+// enable `assert` in RELEASE
+#undef NDEBUG
+#include <assert.h>
+#define NDEBUG
+#endif
+#include <assert.h>
+
+#ifdef MAM2_DEBUG
```

Comment:
Since only file `mam/v2/mss.c` and `mam/v2/tests/test_mss.c` use `dbg_printf`, the macro should be eliminated for further cleanup.